### PR TITLE
[IIIF-670] Update with big structural changes

### PR DESCRIPTION
**Pull request by ksclarke**
* Change Canvas' so that they contain AnnotationPages wrapping Annotations [IIIF-668]
* Remove Sequence and flesh out Ranges more so that they can provide structure [IIIF-670]
* Rename specific structures to 'items' with types for each specific structure [IIIF-652]
* Update Layer to AnnotationCollection and AnnotationList to AnnotationPage [IIIF-675]
* Update various resources to use ResourceTypes for their type values (instead of type field)
* Move ResourceTypes out of the `utils` package and into the base package
* Add detailed information for exceptions that don't currently supply any
* Create NavigableResource for supplying navDate on Canvas, Collection, Manifest, Range
* Improve Canvas serialization (NON_DEFAULT on w/h and better w/h handling in toJSON())
* Improve the way contexts are handled on manifest (better support for additional ones)
* Create SpecificResource class for embeddng in Range(s) (along with Canvas and Range)
* Create different classes for StartCanvas and StartSelector for more control over start's use
* Rename Collection.Manifest to Collection.Item to be consistent with Collection's "items"
* Improvement and additions to tests

### CODE REVIEW

File: src/test/java/info/freelibrary/iiif/presentation/SpecificResourceTest.java [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+        assertEquals(ResourceTypes.SPECIFIC_RESOURCE, new SpecificResource(myID, myOtherID, SELECTOR).getType());
+    }
+
+    /**
+     * Tests {@link SpecificResource#setSelector(Selector) setSelector} method.
+     */
+    @Test
+    public final void testSetSelector() {
+        assertEquals(SELECTOR, new SpecificResource(myID, myOtherID, SELECTOR).getSelector());
+    }
```
Comment (markmatney) [inline src/test/java/info/freelibrary/iiif/presentation/SpecificResourceTest.java:94]: This doesn't test `setSelector`.

File: src/main/java/info/freelibrary/iiif/presentation/OtherContent.java [hunk 2]
```diff
@@ -13,16 +13,14 @@
  */
 public class OtherContent extends Content<OtherContent> {
```
[the remark this answers is not in the document]
Comment (ksclarke) [reply src/main/java/info/freelibrary/iiif/presentation/OtherContent.java:14]: I'd like to keep it for now even though it's not currently referenced anymore. I think it's going to correspond to ImageContent, which gets put onto the AnnotationPage (so I think this is going to be for non-image content put onto the AnnotationPage).

File: src/test/java/info/freelibrary/iiif/presentation/ManifestTest.java [hunk 3]
```diff
@@ -131,10 +131,60 @@ public void setUp() throws IOException {
...
 
-        sequence.addCanvas(canvas1, canvas2);
         myVertx = Vertx.factory.vertx();
     }
 
+    /**
+     * Tests clearing the manifests.
+     */
+    @Test
+    public void testClearManifests() {
```
[the remark this answers is not in the document]
Comment (ksclarke) [reply src/test/java/info/freelibrary/iiif/presentation/ManifestTest.java:141]: Yes, you're right. I was mixing context and manifest up in my head at one point (I don't know why) and it looks like I missed this one.

File: src/test/java/info/freelibrary/iiif/presentation/RangeTest.java [hunk 4]
```diff
@@ -12,30 +16,141 @@
...
+     * Tests the {@link Range#fromJSON(JsonObject) fromJSON} method.
+     */
+    @Test
+    public void testFromJsonSpecificResource() {
+        final JsonObject jsonObj = new JsonObject(SPECIFIC_RESOURCE_ITEM_JSON);
+        assertEquals(jsonObj, Range.fromJSON(jsonObj).toJSON());
+    }
+
+    /**
+     * SPECIFIC_RESOURCE_ITEM_JSON Tests constructing a range.
```
[the remark this answers is not in the document]
Comment (ksclarke) [reply src/test/java/info/freelibrary/iiif/presentation/RangeTest.java:139]: Yes, that looks like a stray CTRL-V.

File: src/test/java/info/freelibrary/iiif/presentation/properties/StartTest.java [hunk 5]
```diff
@@ -30,21 +31,23 @@ public final void testCanvasStart() {
         final String url = myLorem.getUrl();
```
[the remark this answers is not in the document]
Comment (ksclarke) [reply src/test/java/info/freelibrary/iiif/presentation/properties/StartTest.java:31]: I did debate this for a second when doing it. I think the case could be made to separate them, but I'm going to leave as is (with different methods testing each of the aspects) for now unless you feel strongly about splitting them.

[the remark this answers is not in the document]
Comment (markmatney) [reply src/test/java/info/freelibrary/iiif/presentation/properties/StartTest.java:31]: Fine with me

File: src/main/java/info/freelibrary/iiif/presentation/OtherContent.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (markmatney) [reply src/main/java/info/freelibrary/iiif/presentation/OtherContent.java:14]: Fine with me

**Pull request merged**